BRKey.c

```diff
@@ -284,6 +284,33 @@ size_t BRKeyAddress(BRKey *key, char *addr, size_t addrLen) {
     return addrLen;
 }
 
+// writes the pay-to-witness-pubkey-hash nested in pay-to-script-hash (P2SH-P2WPKH) bitcoin address for key to addr:
+// https://bitcoincore.org/en/segwit_wallet_dev/#creation-of-p2sh-p2wpkh-address
+// returns the number of bytes written, or addrLen needed if addr is NULL
+size_t BRKeyWitnessAddress(BRKey *key, char *addr, size_t addrLen) {
+    // Only create witness addresses of compressed public keys
+    assert(key->compressed);
+
+    UInt160 pubKeyHash = BRKeyHash160(key);
+    uint8_t data[21], redeemScript[1 + BRScriptPushData(NULL, 0, pubKeyHash.u8, sizeof(pubKeyHash))];
+
+    if (! UInt160IsZero(pubKeyHash)) {
+        redeemScript[0] = OP_0;
+        BRScriptPushData(&redeemScript[1], sizeof(redeemScript) - 1, pubKeyHash.u8, sizeof(pubKeyHash));
```

Comment:
@gubatron Here's where it's called, funnily enough, only if the pubKeyHash is non empty.

Here you can read more of what's going on https://bitcoincore.org/en/segwit_wallet_dev/#creation-of-p2sh-p2wpkh-address

```
The P2SH redeemScript is always 22 bytes. It starts with a OP_0, followed by a canonical push of the keyhash (i.e. 0x0014{20-byte keyhash})
```